front/src/App.js

```diff
@@ -156,13 +146,24 @@ class App extends Component {
    * @param onIncorrectToken Callback from AuthenticationScreen to update the
    * AuthenticationScreen in case the token was incorrect.
    */
-  onSubmit(token, onIncorrectToken) {
-    //TODO: Implement a correct version
-    if (token === "techunited") {
-      this.setState({ loggedIn: true, token: token });
-    } else {
-      onIncorrectToken();
-    }
+  AuthSubmit(token, onIncorrectToken) {
+    const l = window.location;
+    console.log(`send authorization to ${l.protocol}//${l.host}/api/v1/auth`);
+    fetch(`${l.protocol}//${l.host}/api/v1/auth`, {
+      method: "GET",
+      headers: new Headers({
+        Authorization: "Basic " + btoa(`user:${token}`)
+      })
+    }).then(response => {
+      if (!response.ok) {
+        onIncorrectToken(response.status);
+      } else if (response.ok) {
```

Comment:
to response.ok or !response.ok, are there any other options? Also, consider swapping the branches around:

```
if (response.ok) {
  response.text().then(result => {
     this.setState({ loggedIn: true, session: result });
     this.connect();
   });
} else {
  onIncorrectToken(response.status);
}
```